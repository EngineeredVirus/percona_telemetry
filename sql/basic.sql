--
-- basic sanity
--

CREATE EXTENSION percona_telemetry;

SELECT name FROM pg_settings WHERE name LIKE 'percona_telemetry.%';

SELECT percona_telemetry_version();
SELECT waiting_on_agent FROM percona_telemetry_status();

DROP EXTENSION percona_telemetry;