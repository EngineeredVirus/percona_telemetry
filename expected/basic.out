--
-- basic sanity
--
CREATE EXTENSION percona_telemetry;
-- Sleep so that waiting on agent becomes true
SELECT pg_sleep(3);
 pg_sleep 
----------
 
(1 row)

SELECT name FROM pg_settings WHERE name LIKE 'percona_telemetry.%';
                 name                  
---------------------------------------
 percona_telemetry.enabled
 percona_telemetry.pg_telemetry_folder
 percona_telemetry.scrape_interval
(3 rows)

SELECT percona_telemetry_version();
 percona_telemetry_version 
---------------------------
 1.0
(1 row)

SELECT regexp_replace(output_file_name, '\d{6,}', '<INSTANCE ID>', 'g') AS output_file_name,
        (last_file_processed IS NOT NULL) AS last_file_processed,
        waiting_on_agent FROM percona_telemetry_status();
            output_file_name            | last_file_processed | waiting_on_agent 
----------------------------------------+---------------------+------------------
 ./percona_telemetry-<INSTANCE ID>.json | t                   | t
(1 row)

DROP EXTENSION percona_telemetry;
